integration/smartbft/smartbft_test.go

```diff
@@ -1577,6 +1572,94 @@ var _ = Describe("EndToEnd Smart BFT configuration test", func() {
+				)
+				ordererRunners = append(ordererRunners, runner)
+				proc := ifrit.Invoke(runner)
+				ordererProcesses = append(ordererProcesses, proc)
+				Eventually(proc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+			}
+
+			peerGroupRunner, _ := peerGroupRunners(network)
+			peerProcesses = ifrit.Invoke(peerGroupRunner)
+			Eventually(peerProcesses.Ready(), network.EventuallyTimeout).Should(BeClosed())
+			peer := network.Peer("Org1", "peer0")
+
+			joinChannel(network, channel)
+
+			By("Waiting for followers to see the leader")
+			Eventually(ordererRunners[1].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+			Eventually(ordererRunners[2].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+			Eventually(ordererRunners[3].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Message from 1"))
+
+			By("Joining peers to testchannel1")
+			network.JoinChannel(channel, network.Orderers[0], network.PeersWithChannel(channel)...)
+
+			By("Deploying chaincode")
+			deployChaincode(network, channel, testDir)
+
+			By("querying the chaincode")
+			sess, err := network.PeerUserSession(peer, "User1", commands.ChaincodeQuery{
+				ChannelID: channel,
+				Name:      "mycc",
+				Ctor:      `{"Args":["query","a"]}`,
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess, network.EventuallyTimeout).Should(gexec.Exit(0))
+			Expect(sess).To(gbytes.Say("100"))
+
+			By("invoking the chaincode")
+			invokeQuery(network, peer, network.Orderers[1], channel, 90)
+
+			numberKill := 3
+			orderer := network.Orderers[numberKill]
+			By(fmt.Sprintf("Killing %s", orderer.Name))
+			ordererProcesses[numberKill].Signal(syscall.SIGTERM)
+			Eventually(ordererProcesses[numberKill].Wait(), network.EventuallyTimeout).Should(Receive())
+
+			Eventually(ordererRunners[0].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("Closing: connection error: desc = \"error reading from server: EOF\""))
+			for i := 0; i < 7; i++ {
+				Eventually(ordererRunners[0].Err(), network.EventuallyTimeout, time.Second).Should(gbytes.Say("pickfirstBalancer: UpdateSubConnState.*IDLE connection error"))
+			}
+
+			By(fmt.Sprintf("Launching %s", orderer.Name))
+			runner := network.OrdererRunner(orderer)
+			runner.Command.Env = append(
+				runner.Command.Env,
+				"FABRIC_LOGGING_SPEC=orderer.consensus.smartbft=debug:grpc=debug",
+				"ORDERER_GENERAL_BACKOFF_MAXDELAY=3s",
+			)
+			ordererRunners[numberKill] = runner
+			proc := ifrit.Invoke(runner)
+			ordererProcesses[numberKill] = proc
+			Eventually(proc.Ready(), network.EventuallyTimeout).Should(BeClosed())
+
+			By("Waiting for followers to see the leader, again")
+			Eventually(ordererRunners[3].Err(), 15*time.Second, time.Second).Should(gbytes.Say("Message from 1 channel=testchannel1"))
+
+			By("invoking the chaincode, again")
+			invokeQuery(network, peer, network.Orderers[2], channel, 80)
```

Comment:
```suggestion
			invokeQuery(network, peer, network.Orderers[numberKill], channel, 80)
```

Probably this will make sure the node is getting up to speed after being restarted.

Reply:
This will show that there is a connect from node 3 to node 0. But not the other way around. And in the test, we are trying to break the connectivity from node 0 to node 3.

But I agree with you, it's fairer to send the transaction to node 3.

Reply:
fixed it